### PR TITLE
Model listing util

### PR DESCRIPTION
**Pull request by huwshimi**
Move the model listing code to a util so it can be shared. This is required by #1652.

### CODE REVIEW

**Comment by huwshimi** [comment]
QA:
- load the profile
- your models should load

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/5084/
Test PASSed.

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/5085/
Test PASSed.

**Comment by kadams54** [comment]
QA is :ok: and :+1: on the code. Always like PRs that result in a net removal of code. Thanks!

File: jujugui/static/gui/src/app/components/user-profile/test-user-profile.js [hunk 1]
```diff
@@ -584,95 +566,20 @@ describe('UserProfile', () => {
...
         getDiagramURL={sinon.stub()}
-        gisf={false}
         interactiveLogin={true}
         pluralize={sinon.stub()}
         storeUser={sinon.stub()}
         user={users.charmstore} />);
     assert.equal(list.callCount, 2);
   });
 
-  it('requests jem envs if jem is provided and updates state', () => {
```
Comment (hatched) [inline jujugui/static/gui/src/app/components/user-profile/test-user-profile.js]: I don't see any tests added here which 'assert' that the `listModels` prop was called with the appropriate values for these three cases.

File: jujugui/static/gui/src/app/views/utils.js [hunk 2]
```diff
@@ -1955,6 +1955,51 @@ YUI.add('juju-view-utils', function(Y) {
...
+    @param {Function} callback The function to be called once the model list
+      list has been received.
+    @returns {Object} the XHR request.
+  */
+  utils.listModels = function(env, jem, user, gisf, callback) {
+    // If gisf is enabled then we won't be connected to a model to know
+    // what facades are supported but we can reliably assume it'll be Juju 2
+    // or higher which will support the necessary api calls.
+    if (!gisf) {
+      if (!env ||
```
Comment (hatched) [inline jujugui/static/gui/src/app/views/utils.js]: If you wanted to get rid of this outer conditional...
```JavaScript
if (!gisf && (!env || ... )) {
```

**Comment by hatched** [comment]
Nice improvement, thanks!
:+1: With a/some tests to make sure `listModels` is called properly from the profile.

File: jujugui/static/gui/src/app/components/user-profile/test-user-profile.js (hunk 1 above)
Comment (huwshimi) [inline jujugui/static/gui/src/app/components/user-profile/test-user-profile.js]: With this update the listModel is only provided a callback. The callback is called a passes through the correct data in this test: https://github.com/juju/juju-gui/pull/1668/files#diff-d56b204f2e7fc3ad8a502c26d7aa20fcR238

Comment (huwshimi) [inline jujugui/static/gui/src/app/components/user-profile/test-user-profile.js]: Oh and I've moved the specific jem/controller/default tests to https://github.com/juju/juju-gui/pull/1668/files#diff-f609bae29c5075867e5e0facee90cdd8R1435

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/5103/
Test PASSed.

**Comment by huwshimi** [comment]
Thanks @kadams54 and @hatched :shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/juju-gui-merge

**Pull request merged**